### DISCUSSION TITLE
AVRDUDE Logo?

### DISCUSSION THREAD

**Discussion by ElectronicsArchiver** [avrdudes/avrdude#841] [Show and tell]
I noticed that the [`avrdudes`] organization doesn't
have a logo and neither does the original project?

*Considering that visual representations are vital for users to associate*
*content, even only having a projects name as it's logo can be beneficial.*

To this end I wanted to try my hand at creating a logo from the `AVRDUDE` name.

![512_AVRDUDE](https://user-images.githubusercontent.com/85485984/149642353-04a5e509-c065-4c7e-a76d-53d3ce260283.png)

*Sorry not sorry for people with light mode.*

*On that note, one would of course use a*
*background shape with a non-black / non-white*
*color for avatars like the Github org logo.*

Please correct me if there is a logo / comments on the design.

[`avrdudes`]: https://github.com/avrdudes

**Comment by mariusgreuel** [comment]
I was playing with logos also, but its a WIP that I think is not ready.

Frankly, your logo differs a lot from what I had in mind, because I do not see any association with AVR technology.

Here is my WIP:
![avrdudes-logo-mono-256](https://user-images.githubusercontent.com/16627208/149813314-7d27700a-c30f-4c61-b1e1-fac7bba52ef5.png)

**Comment by MCUdude** [comment]
Logos are tricky. I like logos that aren't way to over-simplified (which seems to be an ongoing trend), and a logo that "shows" more than just a name. To me the name "Avrdude" means the following to me:

* Really old, but in a good way. Been around for ever, tested and proven
* Open source and available on most platforms
* Supports pretty much every AVR you can think of

I have no idea what the logo should look like, but I like that the AVR letters stands out in @mariusgreuel's draft.

**Reply by ElectronicsArchiver** [reply]
> Supports pretty much every AVR you can think of

You could always use a chip like background to associate it I guess.

<br>

> Really old, but in a good way. Been around for ever, tested and proven

Rust like texture on the outside that is falling of like paint to reveal shiny metal underneath.

<br>

> Open source and available on most platforms

I don't that's something that can be easily conveyed in a logo.

**Reply by bd34n** [reply]
What about a simple chip background with a lighting bolt worked in there to represent the "flashing" part - which is avrdude's primary function?

We're going to get all kinds of preferences - my preference is simple, easy to draw, easy to recognize. Maybe even easy enough for an ascii art splash text.

EDIT: And the name could be worked in there, too - just simple all caps AVRDUDE or something. Below, to the side, above, - this is where a graphic designer would be good.

I'm not really great with gimp or photoshop - the art part of it, I can use those tools just fine. But doing artwork is not my forte'. So I apologize for not providing a potential submission. :-)

**Reply by bd34n** [reply]
I will say that I'm not a fan of turning the "dude" part of avrdude into "dudes" - implying dudes that work on AVR's, and seems a bit sexist, and more importantly, has nothing to do with the software. The "dude" part of avrdude never implied a person, it was a rather forced acronym coming from Downloader/UploaDEr that I came up with after changing the name from 'avrprog' which I thought may someday be a conflict issue with (then) Atmel's avrprog.exe in Atmel Studio. And even though our repo folder is called avrdudes, that's mostly because 'avrdude' was already taken and the owner of the 'avrdude' account could not be contacted. I certainly don't think our logo should have it.

**Reply by dl8dtl** [reply]
> What about a simple chip background with a lighting bolt worked in there

Seomething like that?

![avrdude](https://user-images.githubusercontent.com/8264914/150368589-87ea864c-c240-4a73-a1d0-2eb3c701d180.png)

**Reply by bd34n** [reply]
Yes, in concept, that's exactly what I was thinking of. It has the name, and a simple graphic that represents what it does.

**Comment by dl8dtl** [comment]
Or more like that one ;-)
![avrdude](https://user-images.githubusercontent.com/8264914/150378163-a1f23626-e14d-4c9a-a447-6ff8673ab308.png)

**Comment by bd34n** [comment]
Either - probably the second one. Again ... just some ideas, certainly open to other things.

Thanks, @dl8dtl for putting the concept into something tangible vs a verbal description.

**Comment by dl8dtl** [comment]
![avrdude-3](https://user-images.githubusercontent.com/8264914/150425233-80fa70a5-dae8-4687-9e8c-5ff0bb9d2c36.png)
Oh, and another one, based on real photos. :)

**Reply by bd34n** [reply]
Nice! Maybe see if you can you the same font that @mariusgreuel used - I love that font, looks very cool.

**Comment by dl8dtl** [comment]
OK, @mariusgreuel can you tell me which font it is?
My work is a Gimp design, I could share it if desired. The AVR photo is my own, the lightning originates from a Pixabay photo which explicitly did not impose any restrictions, but I modified it heavily so the resulting artwork is my own anyway. Thus, from a copyright/ownership perspective, I don't see any issues.

**Reply by mariusgreuel** [reply]
The font I used is this one: https://typodermicfonts.com/conthrax/

**Reply by dl8dtl** [reply]
> The font I used is this one: https://typodermicfonts.com/conthrax/

Thanks!

**Reply by bd34n** [reply]
Dang you, @mariusgreuel , now I'm out $29 USD because I had to have that font! :-)

**Reply by dl8dtl** [reply]
ROTFL :-)

**Comment by bd34n** [comment]
If it wouldn't be copyright infringement, a little cloaked "Star Wars" dark-side emperor guy with lighting bolts coming out of his hands into the chip would be a nice touch. :-)

**Comment by MCUdude** [comment]
Here's an other idea. 

<img width="573" alt="image" src="https://user-images.githubusercontent.com/3238730/150434658-91711868-dfde-4dae-bf58-98fa1839cf83.png">

**Reply by ElectronicsArchiver** [reply]
Made another version~
![AVRDUDEv2](https://user-images.githubusercontent.com/85485984/150453757-eb0d6321-ca87-40a0-9ea3-1d3d6a3002ee.png)

**Comment by bd34n** [comment]
That looks great!

**Comment by ElectronicsArchiver** [comment]
Btw, I'd like to remind you to keep the circular limit in mind,
depending on where the logo is used, it will be clipped with
a circle and while you can scale it down to fit, this may make
it look a lot worse and inconsistent.

**Reply by dl8dtl** [reply]
> depending on where the logo is used, it will be clipped with
> a circle

Good point!

**Comment by dl8dtl** [comment]
![avrdude-3](https://user-images.githubusercontent.com/8264914/150477048-1ed050e3-e7b6-4b7a-a983-056ee1c5ff6b.png)
Just to see where it would clip, that's my version with a circle around.

**Comment by MCUdude** [comment]
Here's a different approach. Not designed by me, but by a college that knows his stuff. It's pixel corrected, so it scales nicely, and there are multiple variants. There's also an inverted veriantThe A and V in "AVR symbolizes the upload and download functionality.

| AVR 48x48 <br/>non-inverted | AVR 96x96 <br/>non-inverted | AVR 192x192 <br/>non-inverted |
|-------------|-------------|----------------|
| ![avrdude_icon_48](https://user-images.githubusercontent.com/3238730/150522751-e16bcb92-4ab8-41b6-987e-c9816331d14c.png) | ![avrdude_icon_96](https://user-images.githubusercontent.com/3238730/150522878-b289775d-0cd3-422f-b2d4-2c2599761a14.png) | <img width="192" alt="avrdude_icon_192" src="https://user-images.githubusercontent.com/3238730/150522962-87205da0-6b11-4257-816a-d1b335ef43fe.png"> |

| AVR<br/>DUDE<br/>48x48 <br/>non-inverted | AVR<br/>DUDE<br/> 96x96  <br/>non-inverted| AVR<br/>DUDE<br/> 192x192 <br/>non-inverted |
|-------------|-------------|----------------|
| ![avrdude_icon_word_ver_48](https://user-images.githubusercontent.com/3238730/150523176-bea511be-9cb3-4603-b8e6-f851ec6d0458.png) | ![avrdude_icon_word_ver_96](https://user-images.githubusercontent.com/3238730/150523208-294aff3e-329d-4ee2-9dca-891e1432e633.png) | <img width="192" alt="avrdude_icon_word_ver_192" src="https://user-images.githubusercontent.com/3238730/150523220-eb442ade-d7c0-4960-a80f-0f58d54d4de4.png"> |

| AVRDUDE h48 <br/>non-inverted| AVRDUDE h96 <br/>non-inverted | AVRDUDE h192  <br/>non-inverted|
|-------------|-------------|----------------|
| ![avrdude_icon_word_hor_48](https://user-images.githubusercontent.com/3238730/150523610-515bd7eb-e0fd-410f-979d-91c71fed9600.png) | ![avrdude_icon_word_hor_96](https://user-images.githubusercontent.com/3238730/150523508-b29938cb-af77-48b1-af4f-8987f2cc7b06.png) | <img width="436" alt="avrdude_icon_word_hor_192" src="https://user-images.githubusercontent.com/3238730/150523532-e36e3657-d88e-4db4-aaa2-80b50ee6e78c.png">

<br/>
<br/>

| AVR 48x48 <br/>inverted | AVR 96x96 <br/>inverted | AVR 192x192 <br/>inverted |
|------------------------------|------------------------------|-------------------------------|
| ![avrdude_icon_inv_48](https://user-images.githubusercontent.com/3238730/150530878-7217c168-d7a1-4ca2-8cea-bfc90f3615b0.png) | ![avrdude_icon_inv_96](https://user-images.githubusercontent.com/3238730/150530899-369319cc-5403-4d40-befb-77f47fdde263.png)|<img width="192" alt="avrdude_icon_inv_192" src="https://user-images.githubusercontent.com/3238730/150530989-8e1170e9-6f47-446c-ba62-cf552bf4bcd6.png"> |


| AVR<br/>DUDE<br/>48x48 <br/>inverted | AVR<br/>DUDE<br/> 96x96  <br/>inverted| AVR<br/>DUDE<br/> 192x192 <br/>inverted |
|-------------|-------------|----------------|
| ![avrdude_icon_word_ver_inv_48](https://user-images.githubusercontent.com/3238730/150531340-9fbafe37-dd5a-417d-90ef-4aceec8a6081.png) |![avrdude_icon_word_ver_inv_96](https://user-images.githubusercontent.com/3238730/150531364-cd693630-71b5-4b7c-be41-aeaac259d797.png) | <img width="192" alt="avrdude_icon_word_ver_192" src="https://user-images.githubusercontent.com/3238730/150531397-c7ea80df-7704-4fef-926c-d86922207929.png">|

| AVRDUDE h48 <br/>inverted| AVRDUDE h96 <br/>inverted | AVRDUDE h192  <br/>inverted|
|-------------|-------------|----------------|
| ![avrdude_icon_word_hor_inv_48](https://user-images.githubusercontent.com/3238730/150531515-cd27a3a5-de83-47dd-856b-b0a5c8df228a.png) | ![avrdude_icon_word_hor_inv_96](https://user-images.githubusercontent.com/3238730/150531528-bd84990f-e79f-4047-bb9a-8be7fe173ffe.png) | <img width="436" alt="avrdude_icon_word_hor_inv_192" src="https://user-images.githubusercontent.com/3238730/150531543-e56d017a-2362-4acb-924a-d0f9430a4f37.png"> |

**Reply by bd34n** [reply]
I like those, too ... of those, I'm partial to the middle row with the chip above and the "DUDE" below. Probably would clip circularly better, too - as @ElectronicsArchiver reminded us.

**Reply by ElectronicsArchiver** [reply]
Speaking from a design standpoint, the top and bottom rows are the
clearest, while it may seem like the middle row would be a good fit for
a circular clip, the vertically elongated shape is not well suited for icons
in contrast to the chip design, which is longer on the horizontal axis.

Considering the icon has text in it adds to this even more so.

For icons and circular cutouts I'd recommend the first row and / or 
a variant with the `dude` below the `AVR` ***in the chip itself***, ***without
changing the size of the chip***.

For general use, the third row is clearly well suited, in particular
as a header for the README or other documentation, the
side by side version looks quite fitting.

**Reply by mariusgreuel** [reply]
Kudos to your college, I really like the logo!

I have one suggestion though: The text 'AVR' seems to be transparent - it should probably just be white, so it looks a little nicer when viewed in dark mode.

**Reply by dl8dtl** [reply]
Like this?
<img width="96" alt="150522962-87205da0-6b11-4257-816a-d1b335ef43fe" src="https://user-images.githubusercontent.com/8264914/151440072-2d928ab0-9224-4a92-94fc-7208ec8a2da1.png">

**Reply by mariusgreuel** [reply]
Yes!

**Reply by MCUdude** [reply]
Would the logo look better if all pins looked similar? If/when you decide what to use, I can provide high-res photos.
BTW I'm all ears if you have any feedback regarding the logo on what could be done better/differently. Pins too big or too small? Is the font OK? Design touches missing?

<img width="205" alt="image" src="https://user-images.githubusercontent.com/3238730/151444094-042375dc-47fc-44ed-b96f-4075ef8532d6.png">

**Reply by dl8dtl** [reply]
> Would the logo look better if all pins looked similar?

I'm neutral on that one. Some original DIP carriers indeed had half-cut pins at the edges. My 40-pin AVRs don't, but many smaller AVRs do have them as well.
So far, I've been using the non-"Dude" version for the organization logo, but I can't seem to find anywhere to put a repository logo (for just AVRDUDE) to. I just tried to put it into the Wiki page, they want an image URL, so I'm using the logo URL from this discussion above.

**Reply by dl8dtl** [reply]
Again, a white-filled version of the above, for darkmode browsers.
![150523532-e36e3657-d88e-4db4-aaa2-80b50ee6e78c](https://user-images.githubusercontent.com/8264914/151448190-66336c87-77d9-451d-84ff-1283439eae64.png)

**Reply by MCUdude** [reply]
> I can't seem to find anywhere to put a repository logo (for just AVRDUDE) to

I suggest on very top of the README file. Here are a few examples of other popular repos that have done just this:
https://github.com/bblanchon/ArduinoJson
https://github.com/arendst/Tasmota
https://github.com/esphome/esphome

A high-res image with a white background is ideal, yes

**Reply by dl8dtl** [reply]
Well, RRADME is an idea, too.
![Screenshot_20220127-224134_Firefox_Klar](https://user-images.githubusercontent.com/8264914/151448641-14067251-4727-42ba-9b9c-ea98d025fd68.png)
For a "darkmode" impression, a screenshot from my mobile phone.

**Reply by MCUdude** [reply]
We've done some minor changes to the logo. There's now a white, rounded rectangle as so it works with both light and dark mode themes. The colors have also been adjusted slightly. The AVR icon now has a white background, so it works better as a profile icon.

| AVR icon with white background | AVR icon with dark background |
|-------------------------------------|------------------------------------|
| ![image](https://user-images.githubusercontent.com/3238730/151512932-154d89a4-ce10-4031-b5b7-5b2eabb50d4e.png) | ![image](https://user-images.githubusercontent.com/3238730/151513035-2b4be800-3b32-4e78-a808-c3aa35fafc1b.png) |

| AVRDUDE logo with white background | AVRDUDE logo with dark background |
|----------------------------------------------|----------------------------------------|
| ![image](https://user-images.githubusercontent.com/3238730/151513340-d130eac1-db25-4829-a90b-4f5890ced95b.png) | ![image](https://user-images.githubusercontent.com/3238730/151513219-c6886aa2-ed0f-4ea7-a80b-928eb8ee9260.png) |

I've attached the ZIP file that contains these two logo variants in various sizes and formats:
[avrdude_icon_final.zip](https://github.com/avrdudes/avrdude/files/7957009/avrdude_icon_final.zip)

**Reply by dl8dtl** [reply]
Thanks a lot, I've updated the organization icon and the avrdude Wiki home page.

**Reply by MCUdude** [reply]
Use the files in the attached ZIP-file.
The preview I used in the previous post is just low-res previews just to demonstrate how the would look with light and dark mode

**Reply by dl8dtl** [reply]
I have to put it somewhere so I can create a link in the Wiki.
![avrdude_word_480](https://user-images.githubusercontent.com/8264914/151520555-ef2a7560-7002-4354-8b14-49b0b9fa017d.png)

**Reply by dl8dtl** [reply]
A bit too large now?

**Reply by MCUdude** [reply]
Scale it down like so:

```
<img width="40%" alt="avrdude_icon" src="https://user-images.githubusercontent.com/8264914/151520555-ef2a7560-7002-4354-8b14-49b0b9fa017d.png">
```

<img width="40%" alt="avrdude_icon" src="https://user-images.githubusercontent.com/8264914/151520555-ef2a7560-7002-4354-8b14-49b0b9fa017d.png">

EDIT: use width in percentage instead of pixels, so that dynamic scaling works better on phones and tablets

**Reply by dl8dtl** [reply]
I wouldn't want to hack the Wiki with HTML tags. Here's a scaled (from the SVG) version to 240 px height.
![avrdude_word_240](https://user-images.githubusercontent.com/8264914/151530351-8fd5d521-ed9f-439a-bf6f-83d070307fbc.png)

**Comment by dl8dtl** [comment]
The edited lower rows are almost unreadable for me … probably only work with darkmode browsers.
Otherwise, agree with Brian, the middle row of the top 3 ones looks nice to me, too.

**Reply by bd34n** [reply]
Yep - same. Probably need to work a bit with the colors for more contrast?

**Comment by dl8dtl** [comment]
Update with the font Marius mentioned:
![avrdude-3](https://user-images.githubusercontent.com/8264914/150600818-917fb86c-f495-4353-983a-2b18a51ebdef.png)

**Reply by bd34n** [reply]
I think it must be the shadowed text that is messing with my eyes on that.

**Reply by dl8dtl** [reply]
> I think it must be the shadowed text that is messing with my eyes on that.

OK, I added it later on, since my first impression without it was it looks a little dull.
So one more attempt, with a much lighter shadow:

![avrdude-3](https://user-images.githubusercontent.com/8264914/150602933-ae0c5e43-1052-4364-9c0e-9d9ce135d1c3.png)

But we've got plenty of proposals now anyway :), it's probably time to decide …

**Reply by bd34n** [reply]
Agreed. It's hard ... there are aspects of all of them that are good.

My personal preference is @ElectronicsArchiver 's up-arrow down-arrow chip:

![image](https://user-images.githubusercontent.com/51387795/150603460-0600dda2-9d40-40e5-aec1-d681e59733e7.png)

And changing the "dude" font to something less 1970'ish - and I'm not sure how the "lined" font will translate in resizing, etc. I like that design, though ... good spacing, good contrast - should look good on dark on light background, and overall nice design.

**Reply by dl8dtl** [reply]
Except there are no 16-pin AVRs. :-))
But I agree with you, with a slightly different "dude", I would be fine with that.

**Reply by bd34n** [reply]
Well, I'm partial to the ATmega128 or ATmega2560, so maybe a TQFP-64 or -100.

Now we can debate what chip package to use. :-)

**Reply by dl8dtl** [reply]
ATmega2560 would be, well, interesting. :-)

**Reply by bd34n** [reply]
Too much? :-)

-64 is probably too much too ... the square package of a TQFP might be good, though - it would square-up the logo and work good for that purpose, fit ok in a circle with clipping (as long as the corners aren't cut off. But yeah, I wouldn't want to put all those pins on there. Maybe a mega168 (also an excellent chip) which comes in a TQFP-32 - 8 pins to a side (vs 16 or 25).

Or just go with it as-is (with font update), and claim yes, we know, we did it on purpose. :-)

**Reply by bd34n** [reply]
And maybe ElectronicsArchiver can get that cool font Marius found.

**Reply by dl8dtl** [reply]
> Or just go with it as-is (with font update)

Yes, sure. The remark about the 16-pin AVRs was a joke only.
I completely agree, with the font update, I'm fine with it as is.

**Reply by bd34n** [reply]
Same.

**Reply by MCUdude** [reply]
The idea with the three Avrdude logo variants was the the chip alone could be used for avrdudes (hosting multiple AVR projects), and the other two could be used depending on the desired form factor. Regarding the inverted ones, here's how they'll look on a black background. But regardless of which logo you chose, make sure it's designed so that aliasing doesn't kill it when scaled way down

<img width="508" alt="Skjermbilde 2022-01-21 kl  23 24 08" src="https://user-images.githubusercontent.com/3238730/150608240-07a65145-e6a5-43a4-b5ef-9e3ae046c773.png">

**Reply by dl8dtl** [reply]
> The idea with the three Avrdude logo variants was the the chip alone could be used for avrdudes (hosting multiple AVR projects),

Agreed.

**Reply by MCUdude** [reply]
Here's an example of what I mean by scaling. The black square is 200x200 pixels, and the AVR icon in the middle is 48x48 pixels and the AVR\nDUDE icon is 44x44 but still readable.

<img src="https://user-images.githubusercontent.com/3238730/150610225-b138f67d-e16a-4ae9-9865-8dafad6ca119.png" width="200" height="200" />

<img src="https://user-images.githubusercontent.com/3238730/150610560-8061ac8a-32ff-4248-8a2f-c034a8ed9d3e.png" width="200" height="200" />
